python/pyspark/sql/types.py

```diff
@@ -71,6 +67,15 @@ def json(self):
                           separators=(',', ':'),
                           sort_keys=True)
 
+    def needSerialization(self):
```

Comment:
I guess that this could also be a `@property` so that we can write things like `if any(fieldType.needsSerialization for f in fields)`, but I don't feel strongly about this.